csm_web/scheduler/models.py

```diff
@@ -183,10 +156,41 @@ class Spacetime(models.Model):
         (SATURDAY, "Saturday"),
         (SUNDAY, "Sunday"),
     )
-    location = models.CharField(max_length=100)
-    start_time = models.TimeField()
-    duration = models.DurationField()
-    day_of_week = models.CharField(max_length=2, choices=DAY_OF_WEEK_CHOICES)
+    _location = models.CharField(max_length=100)
+    _start_time = models.TimeField()
+    _duration = models.DurationField()
+    _day_of_week = models.CharField(max_length=3, choices=DAY_OF_WEEK_CHOICES)
+
+    """
+    Unfortunately the Django models.Model class doesn't play nice with standard Python metaprogramming
+    functionality like __getattribute__, and Django doesn't allow you to shadow fields in a descendant class,
+    so we have to resort to this boilerplate @property solution in order to have Spacetime 'magically' return
+    the overriden value if there is one
+    """
```

Comment:
we may want some way to expose the original spacetime and whether or not this has an override, so on the frontend we can avoid confusing people about whether an updated time is weekly or temporary

Reply:
You can still access to original spacetime fields, since Python doesn't _actually_ have private methods/fields and the leading underscore is just a convention. The `has_valid_override` method seems to address your second point. Am I misunderstanding, and if so what other functionality did you want?

Reply:
hm that's true, although i was thinking of having a way to access the original field without breaking that convention

`has_valid_override` is good, but i don't see it in the diff: can you give me a line number?

Reply:
oh just kidding it just showed up